include/kangaru.hpp

```diff
@@ -58,7 +58,9 @@ struct function_traits<R(*)(Args...)> {
 	using argument_types = std::tuple<Args...>;
 };
 
-struct Holder {};
+struct Holder {
+	virtual ~Holder() = default;
```

Comment:
I've added `cout` message to `class D` dtor. And if you remove `virtual` here, you can see that D is never destroyed. And the same is for the other classes in the example.